src/ui/ui_render/bootstrap/app_bootstrap.test.js

```diff
@@ -0,0 +1,138 @@
+import mockFs from 'mock-fs';
+import { resolve } from 'path';
+
+const mockTemplate = `
+{{appId}}
+{{bundlePath}}
+{{i18n 'foo'}}
+`;
+
+const templatePath = resolve(__dirname, 'template.js.hbs');
+
+beforeEach(() => {
+  mockFs({
+    [templatePath]: mockTemplate
+  });
+});
+afterEach(mockFs.restore);
+
+import { AppBootstrap } from './app_bootstrap';
+
+describe('ui_render/AppBootstrap', () => {
+  describe('getJsFile()', () => {
+    test('resolves to a string', async () => {
+      expect.assertions(1);
+
+      const boostrap = new AppBootstrap(mockConfig());
+      const contents = await boostrap.getJsFile();
+
+      expect(typeof contents).toEqual('string');
+    });
+
+    test('interpolates templateData into string template', async () => {
+      expect.assertions(2);
+
+      const boostrap = new AppBootstrap(mockConfig());
+      const contents = await boostrap.getJsFile();
+
+      expect(contents).toContain('123');
+      expect(contents).toContain('/foo/bar');
+    });
+
+    test('supports i18n', async () => {
+      expect.assertions(1);
+
+      const boostrap = new AppBootstrap(mockConfig());
+      const contents = await boostrap.getJsFile();
+
+      expect(contents).toContain('translated foo');
```

Comment:
Why is the translation for the key `foo` in the js file?

Reply:
The key is `foo` in the mock template for this test.  In the actual template, it is `UI-WELCOME_ERROR`, which is the translation key for the error message that is shown when bundles fail to load properly: https://github.com/elastic/kibana/blob/master/src/ui/ui_render/views/ui_app.jade#L143